### DISCUSSION TITLE
Configuring PHP directives without custom images

### DISCUSSION THREAD

**Discussion by pompushko** [docker-library/wordpress#879] [General]
Hello

Is there any way to increase upload max file size (upload_max_filesize) in PHP.INI without creating own image?
Why in in 2k24 by default only 2mb?
Is it possible to use environment variables?

Thank you

**Comment by tianon** [comment]
The default value comes from PHP's own defaults (we very intentionally do not modify upstream defaults if/when we can help it).

Whether you can do this without creating an image depends a little bit on how/what you're running.  If you're running the PHP CLI, `-d` is the flag you're looking for (`-d foo[=bar]     Define INI entry foo with value 'bar'`).

If you're running Apache or FPM, you're going to have to do something more involved (specifically, creating/using a `php.ini` file).  The good news is that you can reference environment variables in `php.ini` and they'll be interpreted at runtime, and you can also use a complex command to avoid creating an image, if you just need this in a pinch: `docker run ... php:...-apache bash -Eeuo pipefail -c 'echo foo=bar >> /usr/local/etc/php/php.ini && exec docker-entrypoint.sh apache2-foreground'`

**Comment by pompushko** [comment]
> The default value comes from PHP's own defaults (we very intentionally do not modify upstream defaults if/when we can help it).
> 
> Whether you can do this without creating an image depends a little bit on how/what you're running. If you're running the PHP CLI, `-d` is the flag you're looking for (`-d foo[=bar] Define INI entry foo with value 'bar'`).
> 
> If you're running Apache or FPM, you're going to have to do something more involved (specifically, creating/using a `php.ini` file). The good news is that you can reference environment variables in `php.ini` and they'll be interpreted at runtime, and you can also use a complex command to avoid creating an image, if you just need this in a pinch: `docker run ... php:...-apache bash -Eeuo pipefail -c 'echo foo=bar >> /usr/local/etc/php/php.ini && exec docker-entrypoint.sh apache2-foreground'`


Thank you. But if I will use clean Wordpress official image in AWS ECS? Now it works perfect from the box, without any changes....
For such case, I should create own repository, I should create new pipeline to build container with a new Docker file... And not only.... And thats only for use official image with increased **upload_max_filesize** variable :)

**Comment by pompushko** [comment]
Dear developers, please, increase the limit in image. Thank you :)

**Comment by pompushko** [comment]
@tianon  could you please, advise other solutions?
Wordpress its a good product which used widely, also, many uses of that, upload files more than 2mb :)
And I still cant understand, why such simple things cannot be fixed :)
